server.go

```diff
@@ -594,10 +595,7 @@ func newServer(listenAddrs []net.Addr, chanDB *channeldb.DB,
 	// automatically create an onion service, we'll initiate our Tor
 	// controller and establish a connection to the Tor server.
 	if cfg.Tor.Active && (cfg.Tor.V2 || cfg.Tor.V3) {
-		s.torController = tor.NewController(
-			cfg.Tor.Control, cfg.Tor.TargetIPAddress,
-			cfg.Tor.Password,
-		)
+		s.torController = torController
```

Comment:
Nit: we don't need to set this based on the flag anymore since that's already checked within the config parsing, so we can just set it directly when initializing the server struct.